lib/http/2/compressor.rb

```diff
@@ -549,7 +550,16 @@ def header(buf)
       # @return [Array] +[[name, value], ...]+
       def decode(buf)
         list = []
-        list << @cc.process(header(buf)) until buf.empty?
+        decoding_pseudo_headers = true
```

Comment:
As a brief but related aside...

> Pseudo-header fields MUST NOT appear in trailers. Endpoints MUST treat a request or response that contains undefined or invalid pseudo-header fields as malformed (Section 8.1.2.6).

Proposed update does not break trailer support, but we're also not enforcing the above variant. At a minimum, perhaps worth flagging as a TODO. That said, this also probably does not belong in Compressor, but should be handled in the stream, as that's the right place to track state for initial set of headers vs trailers.

Reply:
You're right.  I think it needs to be opened as a separate issue, though.  There may be several other similar MUST requirements that are not being enforced.  For example, in Section 8.1.1, it says: 

> Pseudo-header fields are only valid in the context in which they are defined. Pseudo-header fields defined for requests MUST NOT appear in responses; pseudo-header fields defined for responses MUST NOT appear in requests. Pseudo-header fields MUST NOT appear in trailers. Endpoints MUST treat a request or response that contains undefined or invalid pseudo-header fields as malformed (Section 8.1.2.6).

For example, you cannot include a `:status` pseudo header in a request nor a `:method` pseudo header in a response.  As far as I can tell, the gem does not enforce this requirement either.  My recommendation is that we create an overarching issue that allows other interested contributors to comment on header validation so that we can gather input from the community and prioritize how we proceed/divide up the work.  What do you think?